File: C/Algorithms/1491.c
#include <stdlib.h>

int cmpi(const void *x, const void *y) {
  int a = *(const int *)x;
  int b = *(const int *)y;

  return (a > b) - (a < b);
}

double average(int *salary, int salarySize) {
  int count = salarySize - 2;
  qsort(salary, salarySize, sizeof(*salary), cmpi);
  int sum = 0;
  for (int i = 1; i < count + 1; i++) {
    sum += salary[i];
  }
  return (double)sum / (double)count;
}

File: Rust/Algorithms/1491.rs
impl Solution {
    pub fn average(mut salary: Vec<i32>) -> f64 {
        let div = f64::from((salary.len() - 2) as i32);
        salary.sort_unstable();
        let sum = *&salary[1..salary.len() - 1].iter().fold(0, |acc, x| acc + x);
        f64::from(sum) / div
    }
}
